.github/workflows/mac_unit_test_ci.yml

```diff
@@ -13,5 +13,5 @@ jobs:
     - uses: actions/checkout@v4
     - name: Build and test
       run: |
-        dotnet build -c Release
+        dotnet build -c Release -warnaserror /p:TreatWarningsAsErrors=true /p:RunAnalyzersDuringBuild=true
```

Comment:
you can simplify this by using `-warnaserror /p:RunAnalyzers=true`

Reply:
https://learn.microsoft.com/en-us/visualstudio/code-quality/disable-code-analysis?view=vs-2022

Reply:
@Piedone can you try removing the flag altogether? Is should be the same result since it's true by default

Reply:
I learned with blood, sweat, and tears (mostly tears) that `/p:TreatWarningsAsErrors=true` is also needed if you want to be sure that it always works every time, everywhere. Perhaps some .NET update fixed that but I wouldn't like to find out that there are suddenly warnings that the build didn't trip on.

Could catch on `RunAnalyzers`, I changed that, it indeed still works as it should.

If we remove the flag then one can disable it in a csproj :). I want to avoid that (and rather configure analyzer suppression selectively and specifically).

OK to merge like this?

Reply:
Yes. We don't any additional tears lol